spec/models/stats/stats_report_spec.rb

```diff
@@ -167,6 +167,64 @@
     end
   end
 
+  describe '#document#path' do
+    let(:report) { create :stats_report, document_file_name: 'test_file.csv' }
+
+    before do
+      stub_const 'REPORTS_STORAGE_PATH', 'reports/:filename'
+    end
+
+    context 'without an Active Storage attachment' do
+      it 'has a path based on the filename' do
+        expect(report.document.path).to eq 'reports/test_file.csv'
+      end
+    end
+
+    context 'with an Active Storage attachment in disk storage' do
+      require 'active_storage/service/disk_service'
+
+      before do
+        ActiveStorage::Attachment.connection.execute(<<~SQL)
+          INSERT INTO active_storage_blobs (key, filename, content_type, metadata, byte_size, checksum, created_at)
+            VALUES('test_key', 'testfile_csv', 100, '{}', 100, 'abc==', NOW())
+        SQL
+        ActiveStorage::Attachment.connection.execute(<<~SQL)
+          INSERT INTO active_storage_attachments (name, record_type, record_id, blob_id, created_at)
+            VALUES('document', 'Stats::StatsReport', #{report.id}, LASTVAL(), NOW())
+        SQL
+
+        service = ActiveStorage::Service::DiskService.new(root: '/root/')
+        allow(ActiveStorage::Blob).to receive(:service).and_return(service)
+      end
+
+      it 'has the path for disk storage' do
+        expect(report.document.path).to eq '/root/te/st/test_key'
+      end
+    end
+
+    context 'with an Active Storage attachment in S3' do
+      require 'active_storage/service/s3_service'
+
+      before do
+        ActiveStorage::Attachment.connection.execute(<<~SQL)
```

Comment:
[as above](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3816/files#r596672810)